### PR TITLE
Get notification email data from salesforce

### PR DESCRIPTION
**Pull request by frj**
This PR adds the fetching fields from salesforce contact objects required to populate the price rise notification email.

These complete list of fields is:
contactID
IdentityID
First Name
Last Name
Address Line 1
Town
Postcode
County
Country
New price
Next payment date
Frequency of bill 

This includes:
- Implementing getContact in the salesforce client
- Mapping of CohortItem and Contact fields to the EmailMessage case class, this is the format of the sqs message that will be sent to membership workflow.
- Creating a no-op logging implementation of the EmailSenderService

### CODE REVIEW

File: lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala [hunk 1]
```diff
@@ -11,27 +12,71 @@ import zio.{Runtime, ZEnv, ZIO, ZLayer, clock, console}
 object NotificationEmailHandler {
   private val NotificationEmailLeadTimeDays = 30
```
Comment (kelvin-chappell) [inline lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala:13]: Perhaps should go in config

File: lambda/src/main/scala/pricemigrationengine/model/Failure.scala [hunk 2]
```diff
@@ -23,3 +23,4 @@ case class S3Failure(reason: String) extends Failure
 case class SubscriptionIdUploadFailure(reason: String) extends Failure
 
 case class NotificationEmailFailure(reason: String) extends Failure
+case class EmailSenderFailure(reason: String) extends Failure
```
Comment (kelvin-chappell) [inline lambda/src/main/scala/pricemigrationengine/model/Failure.scala:26]: What's the difference between these?

File: lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala (hunk 1 above)
Comment (frj) [reply lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala:13]: I thought about that but you need to make a code change to change the config version in the cloudformation template to get any change to the config to take effect. 

So it seems easier to just have the config in code. Its a drawbacks of using cloudformation dynamic references for config.

File: lambda/src/main/scala/pricemigrationengine/model/Failure.scala (hunk 2 above)
Comment (frj) [reply lambda/src/main/scala/pricemigrationengine/model/Failure.scala:26]: Oops

Comment (frj) [reply lambda/src/main/scala/pricemigrationengine/model/Failure.scala:26]: Ah, actually they are different, one is raised by the EmailSender service one by the NotificationEmailHandler.

Ill leave it as is.

File: lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala (hunk 1 above)
Comment (kelvin-chappell) [reply lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala:13]: 👍

**Pull request merged**